docs/aot-jit-switch.md

```diff
@@ -0,0 +1,184 @@
+2    | inactive | candidate | Script v1 | v5.0 | R4.0-AOT-cost, R4.0-AOT-verify
+3    | inactive | candidate | Script v2 | v4.0 | skip-pool-tx (cannot handle)
+4    | inactive | candidate | Script v2 | v5.0 | skip-pool-tx (wait activation)
+||||
+5    | inactive | mined     | Script v1 | v4.0 | R4.0-AOT-cost, R4.0-AOT-verify 
+6    | inactive | mined     | Script v1 | v5.0 | R4.0-AOT-cost, R4.0-AOT-verify 
+7    | inactive | mined     | Script v2 | v4.0 | skip-reject (cannot handle)
+8    | inactive | mined     | Script v2 | v5.0 | skip-reject (wait activation)
+||||
+9    | active   | candidate | Script v1 | v4.0 | R4.0-AOT-cost, R4.0-AOT-verify
+10   | active   | candidate | Script v1 | v5.0 | R5.0-JIT-verify 
+11   | active   | candidate | Script v2 | v4.0 | skip-pool-tx (cannot handle)
+12   | active   | candidate | Script v2 | v5.0 | R5.0-JIT-verify 
+||||
+13   | active   | mined     | Script v1 | v4.0 | skip-accept (rely on majority)
+14   | active   | mined     | Script v1 | v5.0 | R5.0-JIT-verify 
+15   | active   | mined     | Script v2 | v4.0 | skip-accept (rely on majority)
+16   | active   | mined     | Script v2 | v5.0 | R5.0-JIT-verify 
+
+Note the following properties of the validation rules.
+
+1. Rules 1-4 specify creation of new candidate blocks _before_ soft-fork is activated. 
+They require that the behaviour of v4.0 and v5.0 nodes should be identical.
+
+2. Rules 9-10 specify creation of new candidate blocks _after_ soft-fork is activated.
+They are different for v4.0 and v5.0 nodes, but
+[equivalent](#equivalence-properties-of-validation-actions) with respect to preserving
+consensus (see also [Rule Descriptions](#rule-descriptions) for details).
+
+3. For any given tuple (`SF Status`, `Script Version`, `Release`) the _equivalent_ `ValidationAction` is
+applied for both `candidate` and `mined` blocks. This proves the consistency of the rules
+with respect to the change of the block status from `candidate` to `mined`, both before
+and after soft-fork activation.
+
+4. Each rule `i`, where `i` is an odd number, defines a `Validation Action` performed by
+a v4.0 node. Each such rule is paired with the `i+1` rule which defines `Validation Action`
+performed by a v5.0 node. Any such a pair `(i, i+1)` of rules have `Validation Actions`
+which are either the same or equivalent with respect to the [Equivalence Properties of
+Validation Actions](#equivalence-properties-of-validation-actions). This proves
+consistency of validation actions across v4.0 and v5.0 nodes.
+
+5. After SF is activated (`SF Status == active`), both AOT-cost and AOT-verify
+implementations are no longer used in `Validation Action` of v5.0 nodes. The only context
+where v5.0 node needs to use AOT based verification is given by Rule 6, which is to verify 
+a v1 script in a historical mined block before SF is activated. 
+However relying on _Prop 3_ we can replace Rule 6 in a new v5.0.1 release with the
```

Comment:
```suggestion
However relying on _Prop3_ we can replace Rule 6 in a new v5.0.1 release with the
```

Reply:
Prop 3 is not defined before.

Reply:
Changed it into url which refers to the definition.